SingleBrokerDataDeletionTest.shouldNotCompactNotExportedEvents flaky

**Summary**

- How often does the test fail? - top 3 flaky tests for CW 20, CW 21
- Does it block your work? - not really
- Do we suspect that it is a real failure? - unknown

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
Error Message

Segment not open

Stacktrace

java.util.concurrent.ExecutionException: Segment not open
	at io.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:141)
	at io.zeebe.util.sched.future.CompletableActorFuture.get(CompletableActorFuture.java:108)
	at io.zeebe.util.sched.FutureUtil.join(FutureUtil.java:21)
	at io.zeebe.util.sched.future.CompletableActorFuture.join(CompletableActorFuture.java:196)
	at io.zeebe.broker.it.clustering.SingleBrokerDataDeletionTest.shouldNotCompactNotExportedEvents(SingleBrokerDataDeletionTest.java:79)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Caused by: java.lang.IllegalStateException: Segment not open
	at com.google.common.base.Preconditions.checkState(Preconditions.java:508)
	at io.atomix.storage.journal.JournalSegment.checkOpen(JournalSegment.java:234)
	at io.atomix.storage.journal.JournalSegment.createReader(JournalSegment.java:211)
	at io.atomix.storage.journal.SegmentedJournalReader.initialize(SegmentedJournalReader.java:41)
	at io.atomix.storage.journal.SegmentedJournalReader.<init>(SegmentedJournalReader.java:34)
	at io.atomix.storage.journal.SegmentedJournal.openReader(SegmentedJournal.java:227)
	at io.atomix.raft.storage.log.RaftLog.openReader(RaftLog.java:63)
	at io.atomix.raft.partition.impl.RaftPartitionServer.openReader(RaftPartitionServer.java:198)
	at io.zeebe.logstreams.storage.atomix.AtomixRaftServer.create(AtomixRaftServer.java:30)
	at io.zeebe.logstreams.storage.atomix.AtomixReaderFactory.create(AtomixReaderFactory.java:18)
	at io.zeebe.logstreams.storage.atomix.AtomixReaderFactory.create(AtomixReaderFactory.java:22)
	at io.zeebe.logstreams.storage.atomix.AtomixLogStorage.newReader(AtomixLogStorage.java:40)
	at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.<init>(LogStreamReaderImpl.java:41)
	at io.zeebe.logstreams.impl.log.LogStreamImpl.lambda$newLogStreamReader$2(LogStreamImpl.java:117)
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:62)
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39)
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:118)
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107)
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91)
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:204)

</pre>
</details>

**Hypotheses**

**Logs**

<details><summary>Logs</summary>
<pre>
00:16:18.453 [] [main] INFO  io.zeebe.test - Test started: shouldNotCompactNotExportedEvents(io.zeebe.broker.it.clustering.SingleBrokerDataDeletionTest)
00:16:18.458 [] [main] INFO  io.zeebe.test.util.SocketUtil - Choosing next port 27822 for test fork 2 with range PortRange{host='localhost', basePort=27800, maxOffset=100, currentOffset=23}
00:16:18.459 [] [main] INFO  io.zeebe.test.util.SocketUtil - Choosing next port 27823 for test fork 2 with range PortRange{host='localhost', basePort=27800, maxOffset=100, currentOffset=24}
00:16:18.459 [] [main] INFO  io.zeebe.test.util.SocketUtil - Choosing next port 27824 for test fork 2 with range PortRange{host='localhost', basePort=27800, maxOffset=100, currentOffset=25}
00:16:18.459 [] [main] INFO  io.zeebe.test.util.SocketUtil - Choosing next port 27825 for test fork 2 with range PortRange{host='localhost', basePort=27800, maxOffset=100, currentOffset=26}
00:16:18.459 [] [main] INFO  io.zeebe.test.util.SocketUtil - Choosing next port 27826 for test fork 2 with range PortRange{host='localhost', basePort=27800, maxOffset=100, currentOffset=27}
00:16:18.459 [] [main] DEBUG io.zeebe.broker.system - Initializing system with base path /tmp/junit5766860902055733790/0
00:16:18.461 [] [main] INFO  io.zeebe.test.util.SocketUtil - Choosing next port 27827 for test fork 2 with range PortRange{host='localhost', basePort=27800, maxOffset=100, currentOffset=28}
00:16:18.461 [] [main] INFO  io.zeebe.test.util.SocketUtil - Choosing next port 27828 for test fork 2 with range PortRange{host='localhost', basePort=27800, maxOffset=100, currentOffset=29}
00:16:18.461 [] [Thread-559] INFO  io.zeebe.broker.system - Version: 0.24.0-SNAPSHOT
00:16:18.466 [] [Thread-559] INFO  io.zeebe.broker.system - Starting broker 0 with configuration {
  "network" : {
    "host" : "0.0.0.0",
    "portOffset" : 0,
    "maxMessageSize" : "0MB",
    "advertisedHost" : "0.0.0.0",
    "commandApi" : {
      "host" : "0.0.0.0",
      "port" : 27824,
      "advertisedHost" : "0.0.0.0",
      "advertisedPort" : 27824,
      "address" : "0.0.0.0:27824",
      "advertisedAddress" : "0.0.0.0:27824"
    },
    "internalApi" : {
      "host" : "0.0.0.0",
      "port" : 27825,
      "advertisedHost" : "0.0.0.0",
      "advertisedPort" : 27825,
      "address" : "0.0.0.0:27825",
      "advertisedAddress" : "0.0.0.0:27825"
    },
    "monitoringApi" : {
      "host" : "0.0.0.0",
      "port" : 27826,
      "advertisedHost" : "0.0.0.0",
      "advertisedPort" : 27826,
      "address" : "0.0.0.0:27826",
      "advertisedAddress" : "0.0.0.0:27826"
    },
    "maxMessageSizeInBytes" : 8192
  },
  "cluster" : {
    "initialContactPoints" : [ ],
    "partitionIds" : [ 1 ],
    "nodeId" : 0,
    "partitionsCount" : 1,
    "replicationFactor" : 1,
    "clusterSize" : 1,
    "clusterName" : "zeebe-cluster-1",
    "gossipFailureTimeout" : 2000,
    "gossipInterval" : 150,
    "gossipProbeInterval" : 250
  },
  "threads" : {
    "cpuThreadCount" : 2,
    "ioThreadCount" : 2
  },
  "data" : {
    "directories" : [ "/tmp/junit5766860902055733790/0/data" ],
    "logSegmentSize" : "0MB",
    "snapshotPeriod" : "PT1M",
    "logIndexDensity" : 5,
    "logSegmentSizeInBytes" : 8192
  },
  "exporters" : {
    "snapshot-test-exporter" : {
      "jarPath" : null,
      "className" : "io.zeebe.broker.it.clustering.SingleBrokerDataDeletionTest$ControllableExporter",
      "args" : null,
      "external" : false
    }
  },
  "gateway" : {
    "network" : {
      "host" : "0.0.0.0",
      "port" : 27822,
      "minKeepAliveInterval" : "PT30S"
    },
    "cluster" : {
      "contactPoint" : "0.0.0.0:27825",
      "requestTimeout" : "PT15S",
      "clusterName" : "zeebe-cluster",
      "memberId" : "gateway",
      "host" : "0.0.0.0",
      "port" : 27823
    },
    "threads" : {
      "managementThreads" : 1
    },
    "monitoring" : {
      "enabled" : false,
      "host" : "0.0.0.0",
      "port" : 9600
    },
    "security" : {
      "enabled" : false,
      "certificateChainPath" : null,
      "privateKeyPath" : null
    },
    "enable" : false
  },
  "backpressure" : {
    "enabled" : true,
    "algorithm" : "VEGAS"
  },
  "stepTimeout" : "PT5M",
  "executionMetricsExporterEnabled" : false
}
00:16:18.467 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [1/10]: actor scheduler
00:16:18.469 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [1/10]: actor scheduler started in 2 ms
00:16:18.469 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [2/10]: membership and replication protocol
00:16:18.478 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [2/10]: membership and replication protocol started in 9 ms
00:16:18.478 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [3/10]: command api transport
00:16:18.530 [] [Thread-559] DEBUG io.zeebe.broker.system - Bound command API to 0.0.0.0:27824 
00:16:18.531 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [3/10]: command api transport started in 53 ms
00:16:18.531 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [4/10]: command api handler
00:16:18.533 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [4/10]: command api handler started in 1 ms
00:16:18.534 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [5/10]: subscription api
00:16:18.534 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [5/10]: subscription api started in 0 ms
00:16:18.534 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [6/10]: cluster services
00:16:18.556 [] [atomix-0] WARN  io.atomix.primitive.partition.impl.DefaultPartitionGroupMembershipService - Failed to locate partition group(s) via bootstrap nodes. Please ensure partition groups are configured either locally or remotely and the node is able to reach partition group members.
00:16:18.559 [] [atomix-0] INFO  io.atomix.raft.partition.impl.RaftPartitionServer - Starting server for partition PartitionId{id=1, group=raft-partition}
00:16:18.633 [] [raft-server-0-raft-partition-partition-1] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-1} - Transitioning to FOLLOWER
00:16:18.634 [] [raft-server-0-raft-partition-partition-1] INFO  io.atomix.raft.roles.FollowerRole - RaftServer{raft-partition-partition-1}{role=FOLLOWER} - Single member cluster. Transitioning directly to candidate.
00:16:18.634 [] [raft-server-0-raft-partition-partition-1] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-1} - Transitioning to CANDIDATE
00:16:18.634 [] [raft-server-0-raft-partition-partition-1] WARN  io.atomix.utils.event.ListenerRegistry - Listener io.atomix.raft.roles.FollowerRole$$Lambda$387/0x0000000840424840@70419b60 not registered
00:16:18.636 [] [raft-server-0-raft-partition-partition-1] INFO  io.atomix.raft.roles.CandidateRole - RaftServer{raft-partition-partition-1}{role=CANDIDATE} - Single member cluster. Transitioning directly to leader.
00:16:18.645 [] [raft-server-0-raft-partition-partition-1] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-1} - Transitioning to LEADER
00:16:18.646 [] [raft-server-0-raft-partition-partition-1] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-1} - Found leader 0
00:16:18.653 [] [raft-partition-group-raft-partition-0] INFO  io.atomix.raft.partition.RaftPartitionGroup - Started
00:16:18.653 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [6/10]: cluster services started in 118 ms
00:16:18.654 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [7/10]: topology manager
00:16:18.654 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [7/10]: topology manager started in 0 ms
00:16:18.654 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [8/10]: metric's server
00:16:18.657 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [8/10]: metric's server started in 3 ms
00:16:18.657 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [9/10]: leader management request handler
00:16:18.657 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [9/10]: leader management request handler started in 0 ms
00:16:18.657 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 [10/10]: zeebe partitions
00:16:18.658 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 partitions [1/1]: partition 1
00:16:18.658 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Removing follower partition service for partition PartitionId{id=1, group=raft-partition}
00:16:18.659 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Partition role transitioning from null to LEADER
00:16:18.659 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Installing leader partition service for partition PartitionId{id=1, group=raft-partition}
00:16:18.660 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.logstreams.snapshot - Available snapshots: []
00:16:19.582 [] [raft-partition-group-raft-partition-0] INFO  io.atomix.raft.partition.RaftPartitionGroup - Started
00:16:19.635 [] [main] INFO  io.zeebe.gateway - Version: 0.24.0-SNAPSHOT
00:16:19.644 [] [main] INFO  io.zeebe.gateway - Starting gateway with configuration {
  "network" : {
    "host" : "0.0.0.0",
    "port" : 27827,
    "minKeepAliveInterval" : "PT30S"
  },
  "cluster" : {
    "contactPoint" : "0.0.0.0:27825",
    "requestTimeout" : "PT45S",
    "clusterName" : "zeebe-cluster-1",
    "memberId" : "gateway",
    "host" : "0.0.0.0",
    "port" : 27828
  },
  "threads" : {
    "managementThreads" : 1
  },
  "monitoring" : {
    "enabled" : false,
    "host" : "0.0.0.0",
    "port" : 9600
  },
  "security" : {
    "enabled" : false,
    "certificateChainPath" : null,
    "privateKeyPath" : null
  }
}
00:16:19.652 [GatewayTopologyManager] [gateway-scheduler-zb-actors-0] DEBUG io.zeebe.gateway - Received new broker BrokerInfo{nodeId=0, partitionsCount=1, clusterSize=1, replicationFactor=1, partitionRoles={}, partitionLeaderTerms={}, version=0.24.0-SNAPSHOT}.
00:16:19.960 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.logstreams.snapshot - Opened database from '/tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/runtime'.
00:16:19.962 [Broker-0-LogStream-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
00:16:19.962 [Broker-0-StreamProcessor-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.logstreams - Recovering state of partition 1 from snapshot
00:16:19.963 [Broker-0-StreamProcessor-1] [Broker-0-zb-actors-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@791945ba)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@526fb264, configuration: Configuration(false)]
00:16:19.964 [Broker-0-StreamProcessor-1] [Broker-0-zb-actors-1] INFO  io.zeebe.logstreams - Recovered state of partition 1 from snapshot at position -1
00:16:19.965 [Broker-0-StreamProcessor-1] [Broker-0-zb-actors-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@6afeb026)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@59f91c56, configuration: Configuration(false)]
00:16:19.965 [Broker-0-StreamProcessor-1] [Broker-0-zb-actors-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@762a799f)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@15784dde, configuration: Configuration(false)]
00:16:19.969 [Broker-0-Exporter-1] [Broker-0-zb-fs-workers-1] DEBUG io.zeebe.broker.exporter - Recovering exporter from snapshot
00:16:19.970 [Broker-0-HealthCheckService] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - All partitions are installed. Broker is ready!
00:16:19.971 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 partitions [1/1]: partition 1 started in 1313 ms
00:16:19.971 [Broker-0-Exporter-1] [Broker-0-zb-fs-workers-1] DEBUG io.zeebe.broker.exporter - Recovered exporter 'Broker-0-Exporter-1' from snapshot at lastExportedPosition -1
00:16:19.971 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 partitions succeeded. Started 1 steps in 1313 ms.
00:16:19.971 [Broker-0-Exporter-1] [Broker-0-zb-fs-workers-1] DEBUG io.zeebe.broker.exporter - Configure exporter with id 'snapshot-test-exporter'
00:16:19.971 [] [Thread-559] DEBUG io.zeebe.broker.system - Bootstrap Broker-0 [10/10]: zeebe partitions started in 1313 ms
00:16:19.971 [] [Thread-559] INFO  io.zeebe.broker.system - Bootstrap Broker-0 succeeded. Started 10 steps in 1504 ms.
00:16:19.972 [Broker-0-Exporter-1] [Broker-0-zb-fs-workers-1] DEBUG io.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={COMMAND_REJECTION=true, SBE_UNKNOWN=true, COMMAND=true, EVENT=true, NULL_VAL=true}, acceptValueTypes={INCIDENT=true, JOB_BATCH=true, VARIABLE=true, WORKFLOW_INSTANCE_SUBSCRIPTION=true, JOB=true, NULL_VAL=true, WORKFLOW_INSTANCE_CREATION=true, DEPLOYMENT=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, WORKFLOW_INSTANCE_RESULT=true, WORKFLOW_INSTANCE=true, MESSAGE=true, SBE_UNKNOWN=true, MESSAGE_SUBSCRIPTION=true, VARIABLE_DOCUMENT=true, ERROR=true, TIMER=true}}
00:16:19.972 [Broker-0-Exporter-1] [Broker-0-zb-fs-workers-1] DEBUG io.zeebe.broker.exporter - Open exporter with id 'snapshot-test-exporter'
00:16:20.058 [GatewayTopologyManager] [gateway-scheduler-zb-actors-0] DEBUG io.zeebe.gateway - Received metadata change from Broker 0, partitions {1=LEADER} and terms {1=1}.
00:16:20.146 [] [main] INFO  io.zeebe.broker.system - Full replication factor
00:16:20.236 [] [main] INFO  io.zeebe.broker.system - All brokers in topology TopologyImpl{brokers=[BrokerInfoImpl{nodeId=0, host='0.0.0.0', port=27824, version=0.24.0-SNAPSHOT, partitions=[PartitionInfoImpl{partitionId=1, role=LEADER}]}], clusterSize=1, partitionsCount=1, replicationFactor=1, gatewayVersion=0.24.0-SNAPSHOT}
00:16:22.773 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Detected unhealthy components. The current health status of components: {Raft-1=HEALTHY, Broker-0-StreamProcessor-1=UNHEALTHY, logStream=HEALTHY}
00:16:22.774 [Broker-0-HealthCheckService] [Broker-0-zb-actors-1] ERROR io.zeebe.broker.system - Partition-1 failed, marking it as unhealthy
00:16:22.777 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Taking temporary snapshot into /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/pushed-pending/84-1-1588292182776.
00:16:23.004 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Created snapshot for Broker-0-StreamProcessor-1
00:16:23.006 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] INFO  io.zeebe.logstreams.snapshot - Finished taking snapshot, need to wait until last written event position 8589939224 is committed, current commit position is 8589939224. After that snapshot can be marked as valid.
00:16:23.006 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] INFO  io.zeebe.logstreams.snapshot - Current commit position 8589939224 is greater then 8589939224, snapshot is valid.
00:16:23.039 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.broker.clustering.atomix.storage.snapshot.AtomixSnapshotStorage - Purging snapshots older than DbSnapshot{directory=/tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776, metadata=DbSnapshotMetadata{index=84, term=1, timestamp=2020-05-01 12:16:22,776}}
00:16:23.041 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.broker.clustering.atomix.storage.snapshot.AtomixSnapshotStorage - Search for orphaned snapshots below oldest valid snapshot with index 84 in /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/pushed-pending
00:16:23.044 [Broker-0-DeletionService-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.logstreams.delete - Compacting Atomix log up to index 84
00:16:23.045 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.broker.clustering.atomix.storage.snapshot.DbSnapshotStore - Committed new snapshot DbSnapshot{directory=/tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776, metadata=DbSnapshotMetadata{index=84, term=1, timestamp=2020-05-01 12:16:22,776}}
00:16:23.045 [] [raft-server-0-raft-partition-partition-1] DEBUG io.zeebe.broker.clustering.atomix.ZeebeRaftStateMachine - ZeebeRaftStateMachine1{partition=raft-partition-partition-1} - Compacting log up from 45 up to {}
00:16:23.046 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Start replicating latest snapshot /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776
00:16:23.129 [Broker-0-LogStream-1] [Broker-0-zb-actors-0] ERROR io.zeebe.util.actor - Uncaught exception in 'Broker-0-LogStream-1' in phase 'STARTED'. Continuing with next job.
java.lang.IllegalStateException: Segment not open
	at com.google.common.base.Preconditions.checkState(Preconditions.java:508) ~[guava-29.0-jre.jar:?]
	at io.atomix.storage.journal.JournalSegment.checkOpen(JournalSegment.java:234) ~[atomix-storage-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.atomix.storage.journal.JournalSegment.createReader(JournalSegment.java:211) ~[atomix-storage-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.atomix.storage.journal.SegmentedJournalReader.initialize(SegmentedJournalReader.java:41) ~[atomix-storage-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.atomix.storage.journal.SegmentedJournalReader.<init>(SegmentedJournalReader.java:34) ~[atomix-storage-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.atomix.storage.journal.SegmentedJournal.openReader(SegmentedJournal.java:227) ~[atomix-storage-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.atomix.raft.storage.log.RaftLog.openReader(RaftLog.java:63) ~[atomix-cluster-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.atomix.raft.partition.impl.RaftPartitionServer.openReader(RaftPartitionServer.java:198) ~[atomix-cluster-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.logstreams.storage.atomix.AtomixRaftServer.create(AtomixRaftServer.java:30) ~[zeebe-logstreams-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.logstreams.storage.atomix.AtomixReaderFactory.create(AtomixReaderFactory.java:18) ~[zeebe-logstreams-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.logstreams.storage.atomix.AtomixReaderFactory.create(AtomixReaderFactory.java:22) ~[zeebe-logstreams-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.logstreams.storage.atomix.AtomixLogStorage.newReader(AtomixLogStorage.java:40) ~[zeebe-logstreams-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.<init>(LogStreamReaderImpl.java:41) ~[zeebe-logstreams-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.logstreams.impl.log.LogStreamImpl.lambda$newLogStreamReader$2(LogStreamImpl.java:117) ~[zeebe-logstreams-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:62) ~[zeebe-util-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:118) [zeebe-util-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
	at io.zeebe.util.sched.ActorThread.run(ActorThread.java:204) [zeebe-util-0.24.0-SNAPSHOT.jar:0.24.0-SNAPSHOT]
00:16:23.131 [] [main] INFO  io.zeebe.test.records - Test failed, following records where exported:
00:16:23.132 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/000092.sst
00:16:23.133 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/000093.sst
00:16:23.134 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/000094.sst
00:16:23.134 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/000095.sst
00:16:23.135 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/000096.sst
00:16:23.136 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/000097.sst
00:16:23.136 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/CURRENT
00:16:23.137 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/MANIFEST-000004
00:16:23.137 [] [ForkJoinPool.commonPool-worker-5] INFO  io.atomix.raft.partition.RaftPartitionGroup - Stopped
00:16:23.138 [Broker-0-SnapshotDirector-1] [Broker-0-zb-fs-workers-0] DEBUG io.zeebe.logstreams.snapshot - Replicate snapshot chunk /tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/snapshots/84-1-1588292182776/OPTIONS-000090
00:16:23.139 [] [main] DEBUG io.zeebe.gateway - Closing gateway broker client ...
00:16:23.144 [] [main] DEBUG io.zeebe.gateway - topology manager closed
00:16:23.145 [] [main] DEBUG io.zeebe.gateway - Gateway broker client closed.
00:16:23.146 [] [main] DEBUG io.zeebe.broker.system - Closing ClusteringRule...
00:16:23.146 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [1/10]: zeebe partitions
00:16:23.146 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 partitions [1/1]: partition 1
00:16:23.146 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-1] DEBUG io.zeebe.broker.system - Closing Broker-0-Exporter-1
00:16:23.147 [Broker-0-Exporter-1] [Broker-0-zb-fs-workers-1] DEBUG io.zeebe.broker.exporter - Closed exporter director 'Broker-0-Exporter-1'.
00:16:23.147 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - Closed Broker-0-Exporter-1 successfully
00:16:23.147 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - Closing Broker-0-SnapshotDirector-1
00:16:23.147 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - Closed Broker-0-SnapshotDirector-1 successfully
00:16:23.148 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - Closing Broker-0-StreamProcessor-1
00:16:23.148 [Broker-0-StreamProcessor-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-1.
00:16:23.148 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - Closed Broker-0-StreamProcessor-1 successfully
00:16:23.148 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - Closing Broker-0-DeletionService-1
00:16:23.148 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.broker.system - Closed Broker-0-DeletionService-1 successfully
00:16:23.252 [Broker-0-ZeebePartition-1] [Broker-0-zb-actors-0] DEBUG io.zeebe.logstreams.snapshot - Closed database from '/tmp/junit5766860902055733790/0/data/raft-partition/partitions/1/runtime'.
00:16:23.330 [Broker-0-LogStream-1] [Broker-0-zb-actors-0] INFO  io.zeebe.logstreams - Close appender for log stream raft-partition-partition-1
00:16:23.331 [raft-partition-partition-1-write-buffer] [Broker-0-zb-actors-0] DEBUG io.zeebe.dispatcher - Dispatcher closed
00:16:23.331 [Broker-0-LogStream-1] [Broker-0-zb-actors-0] INFO  io.zeebe.logstreams - On closing logstream raft-partition-partition-1 close 3 readers
00:16:23.331 [Broker-0-LogStream-1] [Broker-0-zb-actors-0] INFO  io.zeebe.logstreams - Close log storage with name raft-partition-partition-1
00:16:23.332 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 partitions [1/1]: partition 1 closed in 186 ms
00:16:23.332 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 partitions succeeded. Closed 1 steps in 186 ms.
00:16:23.332 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [1/10]: zeebe partitions closed in 186 ms
00:16:23.332 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [2/10]: leader management request handler
00:16:23.332 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [2/10]: leader management request handler closed in 0 ms
00:16:23.332 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [3/10]: metric's server
00:16:23.337 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [3/10]: metric's server closed in 5 ms
00:16:23.337 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [4/10]: topology manager
00:16:23.338 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [4/10]: topology manager closed in 0 ms
00:16:23.338 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [5/10]: cluster services
00:16:23.338 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [5/10]: cluster services closed in 0 ms
00:16:23.338 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [6/10]: subscription api
00:16:23.338 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [6/10]: subscription api closed in 0 ms
00:16:23.338 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [7/10]: command api handler
00:16:23.339 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [7/10]: command api handler closed in 1 ms
00:16:23.339 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [8/10]: command api transport
00:16:25.558 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [8/10]: command api transport closed in 2218 ms
00:16:25.558 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [9/10]: membership and replication protocol
00:16:25.559 [] [raft-server-0-raft-partition-partition-1] INFO  io.atomix.raft.impl.RaftContext - RaftServer{raft-partition-partition-1} - Transitioning to INACTIVE
00:16:25.634 [] [ForkJoinPool.commonPool-worker-3] INFO  io.atomix.raft.partition.RaftPartitionGroup - Stopped
00:16:27.848 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [9/10]: membership and replication protocol closed in 2290 ms
00:16:27.848 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 [10/10]: actor scheduler
00:16:27.849 [] [main] DEBUG io.zeebe.broker.system - Closing Broker-0 [10/10]: actor scheduler closed in 1 ms
00:16:27.849 [] [main] INFO  io.zeebe.broker.system - Closing Broker-0 succeeded. Closed 10 steps in 4703 ms.
00:16:27.849 [] [main] INFO  io.zeebe.broker.system - Broker shut down.
</pre>
</details>

--- a/qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SingleBrokerDataDeletionTest.java
+++ b/qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SingleBrokerDataDeletionTest.java
@@ -55,6 +55,9 @@ public class SingleBrokerDataDeletionTest {
     // given
     final Broker broker = clusteringRule.getBroker(0);
 
+    final var logstream = clusteringRule.getLogStream(1);
+    final var reader = logstream.newLogStreamReader().join();
+
     while (getSegmentsCount(broker) <= 2) {
       writeToLog();
     }
@@ -74,10 +77,9 @@ public class SingleBrokerDataDeletionTest {
     final var firstSnapshot = clusteringRule.waitForSnapshotAtBroker(broker);
 
     // then
-    final var logstream = clusteringRule.getLogStream(1);
-    final var reader = logstream.newLogStreamReader().join();
     final var firstNonExportedPosition =
         ControllableExporter.NOT_EXPORTED_RECORDS.get(0).getPosition();
+
     reader.seek(firstNonExportedPosition);
     assertThat(reader.hasNext()).isTrue();
     assertThat(reader.next().getPosition()).isEqualTo(firstNonExportedPosition);

